### ISSUE TITLE
[Bug]: Unable to generate a URL for the named route

### ISSUE THREAD

**Issue by haikyuu** [nextcloud/server#39635]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on [Github](https://github.com/nextcloud/server/issues?q=is%3Aopen+is%3Aissue+label%3Abug) OR [Nextcloud Community Forum](https://help.nextcloud.com/) _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

The error happens on a fresh install on a debian cloud image.

```
Unable to generate a URL for the named route \"mail.page.index\" as such route does not exist.
```

### Steps to reproduce

1. I installed next-cloud on a debian cloud image (external maria db + files)
2. I get an internal server error in the ui

### Expected behavior

It should work. The same setup works for a regular debian image. It's automated and the same script does the installation.

Here is a link for the [debian cloud image](https://cloud.debian.org/images/cloud/bookworm/20230723-1450/debian-12-generic-amd64-20230723-1450.qcow2) I used.

And here is [the one that worked](https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-12.0.0-amd64-netinst.iso) 

### Installation method

Community Manual installation with Archive

### Nextcloud Server version

27

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.2

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Fresh Nextcloud Server install

### Are you using the Nextcloud Server Encryption module?

Encryption is Enabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
extcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.example.net",
            "ppd.example.net",
            "172.16.1.9"
        ],
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "version": "27.0.1",
        "overwrite.cli.url": "https:\/\/ppd.example.net",
        "dbtype": "mysql",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "sendmail",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtptimeout": 10,
        "mail_smtpauthtype": "LOGIN",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "maintenance": false,
        "loglevel": 0,
        "theme": "",
        "app_install_overwrite": [
            "files_antivirus",
            "groupfolders",
            "files_downloadactivity",
            "admin_notifications"
        ],
        "updater.release.channel": "stable",
        "mail_sendmailmode": "smtp",
        "encryption.legacy_format_support": false,
        "encryption.key_storage_migrated": false
    }
}
```

### List of activated Apps

```shell
Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade
Enabled:
  - activity: 2.19.0
  - admin_audit: 1.15.0
  - announcementcenter: 6.6.1
  - bruteforcesettings: 2.5.0
  - calendar: 4.4.2
  - circles: 27.0.1
  - cloud_federation_api: 1.10.0
  - comments: 1.15.0
  - contacts: 5.3.2
  - contactsinteraction: 1.6.0
  - dashboard: 7.5.0
  - dav: 1.24.0
  - encryption: 2.13.0
  - federatedfilesharing: 1.15.0
  - federation: 1.17.0
  - files: 1.22.0
  - files_antivirus: 5.1.1
  - files_external: 1.17.0
  - files_pdfviewer: 2.6.0
  - files_rightclick: 1.4.0
  - files_sharing: 1.17.0
  - files_trashbin: 1.15.0
  - files_versions: 1.18.0
  - firstrunwizard: 2.14.0
  - logreader: 2.10.0
  - lookup_server_connector: 1.15.0
  - mail: 2.2.6
  - nextcloud_announcements: 1.14.0
  - notifications: 2.13.1
  - oauth2: 1.15.1
  - password_policy: 1.17.0
  - photos: 2.3.0
  - privacy: 1.9.0
  - provisioning_api: 1.15.0
  - recommendations: 1.4.0
  - related_resources: 1.0.4
  - serverinfo: 1.15.0
  - settings: 1.7.0
  - sharebymail: 1.15.0
  - support: 1.8.0
  - survey_client: 1.13.0
  - systemtags: 1.15.0
  - text: 3.6.0
  - theming: 2.0.1
  - twofactor_backupcodes: 1.14.0
  - updatenotification: 1.15.0
  - user_status: 1.5.0
  - viewer: 1.9.0
  - weather_status: 1.5.0
  - workflowengine: 2.7.0
Disabled:
  - admin_notifications: 1.0.2 (installed 1.0.2)
  - audioplayer: 3.4.0 (installed 3.3.0)
  - files_accesscontrol: 1.17.0 (installed 1.13.1)
  - files_automatedtagging: 1.17.0 (installed 1.13.0)
  - files_texteditor: 2.15.1 (installed 2.14.0)
  - groupfolders: 15.0.1 (installed 11.1.6)
  - ojsxc: 5.0.0 (installed 5.0.0)
  - spreed: 17.0.2 (installed 13.0.6)
  - suspicious_login: 5.0.0
  - twofactor_totp: 9.0.0
  - user_ldap: 1.17.0
  - workspace: 3.0.2
```

### Nextcloud Signing status

```shell
INTERNAL server error, so I can't do this step.
```

### Nextcloud Logs

```shell
{"reqId":"ijbozVvfWAYQYuZOoT0b","level":1,"time":"2023-07-31T10:06:13+00:00","remoteAddr":"88.228.71.4","user":"--","app":"no app in context","method":"GET","url":"/index.php/csrftoken","message":"Unable to generate a URL for the named route \"mail.page.index\" as such route does not exist.","userAgent":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/16.3 Safari/605.1.15","version":"27.0.1","exception":{"Exception":"Symfony\\Component\\Routing\\Exception\\RouteNotFoundException","Message":"Unable to generate a URL for the named route \"mail.page.index\" as such route does not exist.","Code":0,"Trace":[{"file":"/var/www/nextcloud/lib/private/Route/Router.php","line":380,"function":"generate","class":"Symfony\\Component\\Routing\\Generator\\UrlGenerator","type":"->"},{"file":"/var/www/nextcloud/lib/private/URLGenerator.php","line":103,"function":"generate","class":"OC\\Route\\Router","type":"->"},{"file":"/var/www/nextcloud/lib/private/NavigationManager.php","line":320,"function":"linkToRoute","class":"OC\\URLGenerator","type":"->"},{"file":"/var/www/nextcloud/lib/private/NavigationManager.php","line":113,"function":"init","class":"OC\\NavigationManager","type":"->"},{"file":"/var/www/nextcloud/apps/theming/lib/ThemingDefaults.php","line":188,"function":"getAll","class":"OC\\NavigationManager","type":"->"},{"file":"/var/www/nextcloud/lib/private/legacy/OC_Defaults.php","line":280,"function":"getShortFooter","class":"OCA\\Theming\\ThemingDefaults","type":"->"},{"file":"/var/www/nextcloud/lib/public/Defaults.php","line":176,"function":"getLongFooter","class":"OC_Defaults","type":"->"},{"file":"/var/www/nextcloud/core/templates/layout.guest.php","line":52,"function":"getLongFooter","class":"OCP\\Defaults","type":"->"},{"file":"/var/www/nextcloud/lib/private/Template/Base.php","line":180,"args":["/var/www/nextcloud/core/templates/layout.guest.php"],"function":"include"},{"file":"/var/www/nextcloud/lib/private/Template/Base.php","line":150,"function":"load","class":"OC\\Template\\Base","type":"->"},{"file":"/var/www/nextcloud/lib/private/legacy/OC_Template.php","line":179,"function":"fetchPage","class":"OC\\Template\\Base","type":"->"},{"file":"/var/www/nextcloud/lib/private/legacy/OC_Template.php","line":210,"function":"fetchPage","class":"OC_Template","type":"->"},{"file":"/var/www/nextcloud/lib/private/Template/Base.php","line":132,"function":"fetchPage","class":"OC_Template","type":"->"},{"file":"/var/www/nextcloud/lib/private/legacy/OC_Template.php","line":331,"function":"printPage","class":"OC\\Template\\Base","type":"->"},{"file":"/var/www/nextcloud/index.php","line":74,"function":"printExceptionErrorPage","class":"OC_Template","type":"::"}],"File":"/var/www/nextcloud/3rdparty/symfony/routing/Generator/UrlGenerator.php","Line":143,"message":"Unable to generate a URL for the named route \"mail.page.index\" as such route does not exist.","exception":{},"CustomMessage":"Unable to generate a URL for the named route \"mail.page.index\" as such route does not exist."}}
```

### Additional info

This is probably similar to https://github.com/nextcloud/server/issues/31631

**Comment by joshtrichards** [comment]
> Nextcloud or one of the apps require upgrade - only a limited number of commands are available
You may use your browser or the occ upgrade command to do the upgrade

This does not appear to be a fresh install. What do you mean by "external files"?

Does your automation normally trigger an upgrade? Does it generate any logging?

**Comment by cejkato2** [comment]
I had the same issue after upgrading from Nextcloud25 to Nextcloud26 (and further to Nextcloud27); it was on Debian 11, PHP8.1 (FPM), nginx.

Luckily, moving to PHP8.2 solved it for me.

Examples of the error messages I observed in broken version with PHP8.1 after upgrade:

```
Unable to generate a URL for the named route \"mail.page.index\" as such route does not exist.
Unable to generate a URL for the named route \"notes.page.index\" as such route does not exist.
Unable to generate a URL for the named route \"tables.page.index\" as such route does not exist.
Unable to generate a URL for the named route \"cloud_federation_api.RequestHandler.addShare\" as such route does not exist.
```
Generally, I think no URL could be generated.

I'm wondering if there is some kind of cache in PHP or Nextcloud dependencies (such as symfony) that can be cleared and recreated? Maybe there is some incompatibility between different versions that causes URL generation cannot work after upgrade.

**Comment by joshtrichards** [comment]
@cejkato2 Do you think it was 8.2 that fixed it or the act of restarting FPM resulting from upgrading to 8.2? Sometimes we've seen issues due to inconsistencies in PHP's Opcache until the app server (i.e. FPM or apache/mod_php) is bumped (service reloaded).

**Comment by cejkato2** [comment]
Interesting question. I'm pretty sure that I tried to restart both webserver (nginx) and php8.1-fpm service before the upgrade/migration to 8.2.

> Then I switched vhost config of nextcloud to use php8.2-fpm (that I just started) and the error disappeared and nextcloud works normally now, even after several new updates of its Apps.

It's a pity I don't know much about internals of php8.1 caching and its configs - I changed only a few lines based on nextcloud hints/guides, everything else was default by the deb packages. If there is any default system path where php can store something temporal, it would be a nice try to delete it and check again - maybe this could help anybody in the future. Or is there any cache by nexcloud?

By the way, I also tried to download new nextcloud zip from the web and I replaced the whole 3rdparty dir (with symfony etc) removing the original content - this still didn't help for php8.1. Also, I didn't find any significant difference between my local files and the nextcloud downloaded files. Nexcloud source files are stored in `/var/www/nextcloud` but the datadir is set to a different location. Difference between /var/www/nextcloud and files from newly downloaded zip was minimal (except from config, naturally).
Is it possible that some cache dir can be hidden somewhere in the datadir, where the user files are stored? If yes, it was automatically re-created for me after the migration to php8.2. Or is there anything stored in the database? (I use mariadb server)
